### DISCUSSION TITLE
Review your Firebase Cloud Messaging tokens and topic subscriptions

### DISCUSSION THREAD

**Discussion by Alex-collab-byte** [firebase/flutterfire#12668] [Q&A]
I recently received an email regarding Review your Firebase Cloud Messaging tokens and topic subscriptions. I am not sure what I need to do as a Flutter client developer. The general meaning in your email is that tokens and topic subscriptions will become invalid, and obtaining tokens will result in a 404 error message. However, I noticed that your API did not provide an error prompt for obtaining tokens
The email is as follows:
We're writing to inform you that we will be garbage collecting all expired Firebase Cloud Messaging registration tokens and topic subscriptions starting May 15, 2024.

What you need to know
Starting May 15, 2024, the [stale] tokens and topic subscriptions for devices, which have not been connected to FCM in over 270 days, will be considered as expired and invalid. If a request is sent to an invalid token, it will be rejected with a 404 (UNREGISTERED) error.

Regarding topic sends or notification campaigns, it is possible that you may observe a decrease in the time taken for fanouts to complete. This could be attributed to the removal of expired subscriptions and a reduction in the number of messages sent.

This process will help improve fanout performance for your campaigns and identify the undeliverable registration tokens.

What you need to do
We recommend you to confirm that your service handles 404 error codes from FCM correctly. Please [delete your record] of any tokens with a 404 response code because they will not be valid again.

If a device comes back online after the old registration is garbage collected, your application will automatically re-register with FCM and receive a new token. Please follow our [recommendations on ensuring token freshness] to confirm that your app is aware of the token changes and that you update any relevant application servers with this new token for this app instance.

Topic subscriptions will not automatically renew. We recommend you to update your app to re-subscribe to any relevant topics [approximately every month] so that it doesn't get dropped.

You can observe the impact of this garbage collection by monitoring your 404 error rate, overall campaign sent rates, and delivery statistics available to you in our [Aggregate Delivery Data API] offering.

We’re here to help
We understand this may require some planning and we are here to support you. If you have questions, please reach out to our [Firebase support team.]

**Comment by zwu52** [comment]
This would be an no-op from dev side. As said, the 404 will be retuned on/after may 15th
